src/IECore/StreamIndexedIO.cpp

```diff
@@ -3177,10 +3177,12 @@ IndexedIOPtr StreamIndexedIO::directory( const IndexedIO::EntryIDList &path, Ind
 			}
 			else if ( missingBehaviour == IndexedIO::NullIfMissing )
 			{
+				delete newNode;
```

Comment:
Nice catch! Looks like we've got the same bug at [L3175](https://github.com/ImageEngine/cortex/pull/1181/commits/b271c10246400e0c7ebe4aadaa22dbd649af2743#diff-291c8c2556e687258e13f0d34d72624584d8119cba24d7e79a485c2dcbe2b58bR3175), where we're throwing without deleting first? I'd be tempted to modernise this function and do `auto newNode = std::make_unique<StreamIndexedIO::Node>( ... )` at L3160 and `return duplicate( newNode.release() )` at L3189, then we're covered for all code paths without needing manual deletion.

Reply:
Double good catch on the exception! Took your modernization advice in https://github.com/ImageEngine/cortex/pull/1181/commits/f0e640c1a3e6e9c99a524b4b5f74599a1e94b025